src/main/java/org/springframework/data/jpa/repository/query/CriteriaQueryParameterBinder.java

```diff
@@ -60,19 +59,18 @@
 	 */
 	@Override
 	@SuppressWarnings("unchecked")
-	protected void bind(Query query, JpaParameter parameter, Object value, int position) {
+	protected QueryParameterSetter bind(JpaParameter parameter, Object value, Integer position) {
```

Comment:
You are right on both accounts. Will fix once I got feedback from @olivergierke (assuming we are going forward with this approach).